test/functional-tests/pageObjects/ccd-components/fields/ccdCollectionField.js

```diff
@@ -3,13 +3,31 @@ let RandomUtils = require('../../../utils/ccdDataGenerationUtils.js');
 class CcdCollectionField {
 
   constructor(css){
-      this.css = css;
-      this.label = this._getLabel();
-      this.addNewButtonCss = `${this.css} > div > div >button:nth-of-type(1)`;
-      this.lastTextField = `${this.css} > div > div > div > div:last-of-type input`;
-      this.inputValue = null;
-      this.checkYourAnswersValue = '';
+    this.css = css;
+    this.label = this._getLabel();
+    this.button = new Button('.button', 'Add new');
+    this.lastTextField = `${this.css} > div > div > div > div:last-of-type input`;
+    this.inputValue = null;
+    this.checkYourAnswersValue = '';
+  }
 
+  /**
+   * Check if field is ready to type
+   * @returns true or false
+   */
+  async isFieldInputReady(){
```

Comment:
perhaps rename to isButtonReady or isAddNewButtonReady  to make it more clear what element we are checking